lib/matplotlib/font_manager.py

```diff
@@ -1304,16 +1323,51 @@ def findfont(self, prop, fontext='ttf', directory=None,
         rc_params = tuple(tuple(rcParams[key]) for key in [
             "font.serif", "font.sans-serif", "font.cursive", "font.fantasy",
             "font.monospace"])
+
+        if not isinstance(prop, FontProperties):
+            prop = FontProperties._from_any(prop)
+
         return self._findfont_cached(
             prop, fontext, directory, fallback_to_default, rebuild_if_missing,
             rc_params)
 
+    def find_fontsprop(self, prop, fontext='ttf', directory=None,
+                       fallback_to_default=True, rebuild_if_missing=True):
+
+        prop = FontProperties._from_any(prop)
+        ffamily = prop.get_family()
+
+        # maintain two dicts, one for available paths,
+        # the other for fallback paths
+        fpaths, fbpaths = OrderedDict(), OrderedDict()
+        for fidx in range(len(ffamily)):
+            cprop = prop.copy()
+
+            # set current prop's family
+            cprop.set_family(ffamily[fidx])
+
+            try:
+                fpath = self.findfont(cprop, fontext, directory,
+                                      False, rebuild_if_missing)
+                fpaths[ffamily[fidx]] = fpath
+
+            except ValueError as e:
+                if fallback_to_default:
+                    fpath = self.findfont(cprop, fontext, directory,
+                                          True, rebuild_if_missing)
+                    fbpaths.update({self.defaultFamily[fontext]: fpath})
```

Comment:
There is otherwise a nice symmetry between the two code blocks but one updates `fpaths` via assignment and the other updates `fbpaths` via the `update` method. It seems to me they do the same thing, so perhaps this could use assignment as well? It's not a big thing but it catches the eye when scanning the code.

Reply:
The second question is in sync with the first one (and kind of answers it too).
The 'fallback paths' (or `fbpaths`) mostly have the same path (DejaVu Sans) for each family, so, we don't really have to append it multiple times, rather keep changing the same entry. (note it doesn't necessarily have to be DejaVu Sans for each family, but the way `findfont` works right now is to directly get `defaultFamily` for non-found-family)

Reply:
> A completely different question: isn't self.defaultFamily[fontext] going to be constant throughout the function, so the fallback dict will always have at most one element?

Yes on the first part, but no for the latter part. Fallback dict will only exist if one of the font-families were not found while parsing the list.

> If that is the intended result, perhaps it doesn't need to be called inside the loop but just appended after the loop.

That's true, but currently there's no way to find if there was a fallback, because `findfont` calls `findfont` if fallback_to_default is True, see:
https://github.com/matplotlib/matplotlib/blob/c160470aa19263210be00bd22e6940007c6defe4/lib/matplotlib/font_manager.py#L1403-L1416

Reply:
We could add a Try Except clause, but that would mean that the `ValueError` inside `findfont` will never be triggered:

Iterate through the list, call `findfont()` with `fallback_to_default` as False for each. We'll resort to Try Except clause and if a font family is not found, we'll assign a default fallback value and append it at the end of the loop.

Reply:
I think we'll also have to handle the code block I sent in the previous comment.

In a general sense, I designed this to be as less invasive to original API as I could, since it's being used at a lot of places and I didn't want to disturb their workflow.

Reply:
Simplified the logic according to this discussion..